google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -137,7 +137,7 @@ public AppEngineDeployAction createAutoDeploymentAction(
         artifactToDeploy,
         defaultAppYaml(),
         defaultDockerfile(artifactType),
-        AppEngineUtil.generateVersion(),
+        version,
```

Comment:
The only place the version is being generated is in DeployAction. Regardless, you are right its better to confine it to the helper. I'll update it.